### ISSUE TITLE
Is it possible to generate an icc model?

### ISSUE THREAD

**Issue by sangamk** [secure-software-engineering/FlowDroid#42]
Is it possible to generate an ICC model with FlowDroid? Without knowledge of the app so only using the APK? Currently, I have a call graph but I also want to model the communication between activities. Is that possible with FlowDroid?

I saw the option Iccconfiguration which is disabled. What exactly is the IccModel?

My code:
```kotlin    
val app = SetupApplication(androidPlatformPath, appPath)
    app.config.isTaintAnalysisEnabled = false
    app.config.mergeDexFiles = true
    app.config.iccConfig = InfoflowAndroidConfiguration.IccConfiguration() <-- this does not work
    app.constructCallgraph()
```

**Comment by StevenArzt** [comment]
FlowDroid uses ICCTA for inter-component data flow analysis. ICCTA, on the other hand, needs an external ICC model to operate on. You can use IC3 or Harvester to generate such models (or write them by hand). These tools for generating ICC models are not part of FlowDroid. They are built and maintained separately by different people.

Your NoClassDefFoundError seems to be caused by a different problem, though. Are you ysure that you have built FlowDroid with all required dependencies? If you link your own program against FlowDroid, make sure that you include the transitive dependencies as well (should be easy with maven, but take care when building with something else).

**Comment by sangamk** [comment]
Thx for the quick reply. The NoClassDefFoundError was related to a dependency indeed, which I fixed. 
What does this ICC model entail can I find a template for that. I saw that the ICCTA project on GitHub uses IC3 for which the results are pushed to a DB. In Flowdroid (using ICCTA) I see no connection to any db. Is it right for me to assume that the model Flowdroid uses is the result of IC3 which can be found in the IC3 db?
